website/scripts/index_search_content.js

```diff
@@ -1,3 +1,95 @@
-const { indexDocsContent } = require('@hashicorp/react-search/tools')
+require('dotenv').config()
+const fs = require('fs')
+const path = require('path')
+const {
+  indexContent,
+  getDocsSearchObject,
+} = require('@hashicorp/react-search/tools')
+const resolveNavData = require('../components/remote-plugin-docs/utils/resolve-nav-data')
 
-indexDocsContent()
+// Run indexing
+indexContent({ getSearchObjects })
```

Comment:
This indexing approach uses some existing and slightly modified search utilities, which we exposed via https://github.com/hashicorp/react-components/pull/161, to allow us to index content without messing with the filesystem (as we did in our initial approach).

I've testing this indexing with [a temporary Algolia staging index for packer](https://www.algolia.com/apps/YY0FFNI7MF/explorer/browse/staging_PACKER).

> Side note: I would love to get a staging index setup for deploy previews, so we can better test the indexing changes... This is something we have on our `learn` site)... but that's an effort for another day! Going to add this idea to our web platform board.